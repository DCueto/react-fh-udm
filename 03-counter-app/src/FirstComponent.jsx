import PropTypes from 'prop-types';

export const FirstComponent = ( { title, number, subTitle, name } ) => {

  // if( !title ){
  //   console.error('El title no existe');
  // }

  return (
    <div className='card'>
     <h2>{ title }</h2>
     <h3>{ number * 2 }</h3>
     <h4>{ subTitle }</h4>
     <h5>{ name }</h5>
    </div>
  )
}

FirstComponent.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  subTitle: PropTypes.string
}

FirstComponent.defaultProps = {
  title: 'No hay título',
  subTitle: 'No hay subtítulo',
  name: 'Daniel Cueto'
}