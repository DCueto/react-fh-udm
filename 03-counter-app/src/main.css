html, body{
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

.card{
  background-color: #6E8898;
  padding: 20px;
  border-radius: 15px;
  margin: 20px 0;
}

button{
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 5px;
}