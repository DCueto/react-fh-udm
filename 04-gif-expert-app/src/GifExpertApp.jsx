import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'One Punch' ]);

  const onAddCategory = ( newCategory ) => {
    if( categories.includes( newCategory ) ) return;

    setCategories( [ newCategory, ...categories ] );
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        // setCategories={ setCategories }
        onNewCategory = { onAddCategory }
      />

      <section>
        {
          categories.map( (item) =>
            ( <GifGrid
                key={ item }
                category={ item } />
            )
          )
        }
      </section>

    </>
  )

}

export default GifExpertApp;