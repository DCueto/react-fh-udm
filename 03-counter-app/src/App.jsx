import { FirstComponent } from "./FirstComponent";
import { CounterApp } from "./CounterApp";

function App(){
  return (
    <>
      {/* <FirstComponent title='Hola, soy Goku!' number={ 123 } />
      <FirstComponent number={ 123 } /> */}

      <CounterApp value={20} />
    </>
  );
}

export default App;