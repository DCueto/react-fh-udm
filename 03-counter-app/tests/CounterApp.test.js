import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Tests de <CounterApp />', () => {

  const initialValue = 10;

  test('Debe de hacer match con el Snapshot', () => {
    
    const { container } = render(<CounterApp value={30} />);

    expect( container ).toMatchSnapshot();

  });

  test('Debe de mostrar el valor inicial de 100', () => {

    render(<CounterApp value={100} />);

    expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain('100');

  });

  test('Debe de incrementar con el botón +1', () => {

    render( <CounterApp value={initialValue} /> );

    fireEvent.click( screen.getByText('+1') );

    expect( screen.getByText(initialValue + 1) ).toBeTruthy();

  });

  test('Debe de decrementar con el botón -1', () => {

    render( <CounterApp value={initialValue} /> );
    
    fireEvent.click( screen.getByText('-1') );
    // screen.debug();
    expect( screen.getByText(initialValue - 1) ).toBeTruthy();

  });

  test('Debe de funcionar el botón de reset', () => {

    render( <CounterApp value={initialValue} /> );

    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );

    screen.debug();

    // fireEvent.click( screen.getByText('Reset') );
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    screen.debug();
    expect( screen.getByText( initialValue ) ).toBeTruthy();

  });


});