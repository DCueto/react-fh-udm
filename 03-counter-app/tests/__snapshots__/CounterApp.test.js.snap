// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests de <CounterApp /> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h1>
      CounterApp
    </h1>
    <h2>
      30
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
